### PR TITLE
Add Azure Pipelines support

### PR DESCRIPTION
**Pull request by Ky7m**
1. Add integration with Azure Pipelines (I had to disable several tests, it is temporarily)
2. Remove limit for build time on Travis
3. Upgrade xunit packages

### CODE REVIEW

File: azure-pipelines.Ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+trigger:
+  batch: true
+  branches:
+    # blacklist
```
Comment (adamsitnik) [inline azure-pipelines.Ubuntu.yml]: nit: this comment is not required, `excluded` is kind of obvious and enough

File: azure-pipelines.Windows.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+name: $(Year:yyyy).$(Month).$(DayOfMonth)$(Rev:.r)
+
+jobs:
+- template: build/azure-pipelines.job.template.yml
+  parameters:
+    name: Windows
+    vmImage: 'vs2017-win2016'
+    scriptFileName: .\build.ps1
+    timeoutInMinutes: 300
```
Comment (adamsitnik) [inline azure-pipelines.Windows.yml]: why the `timeoutInMinutes` is 300 for Windows when the default value in the template is 120?

Comment (Ky7m) [reply azure-pipelines.Windows.yml]: I need to remove this, but I am still investigating where deadlock is. Usually, it takes 40 minutes to complete build with all test suites on Windows, and in some cases 3 hours isn't enough to finish the build. Only on Windows. macOS and Linux builds are stable.

**Comment by Ky7m** [comment]
@adamsitnik, my pleasure. CoreRT tests are back, pre-requisites were required for Ubuntu agent only.

**Comment by adamsitnik** [comment]
> CoreRT tests are back, pre-requisites were required for Ubuntu agent only

@Ky7m awesome! do we have any blockers now except of the `CustomCoreClrToolchainTests`? Please let me know if you need some help.

**Comment by adamsitnik** [comment]
@Ky7m if you wonder why the builds are not triggered https://blog.github.com/2018-10-21-october21-incident-report/

**Comment by adamsitnik** [comment]
> not clear behavior of test SingleBenchmarkCanBeExecutedForMultipleRuntimes

it's weird, it was rather always stable. The goal of this test is to run the same benchmark for .NET Core and CLR. Were you able to reproduce it locally?

**Comment by Ky7m** [comment]
> it's weird, it was rather always stable. The goal of this test is to run the same benchmark for .NET Core and CLR. Were you able to reproduce it locally?

I guess it is related to fact that tests are run in parallel and it might lead to such behavior. I have a few ideas on how to check it.

**Comment by Ky7m** [comment]
I see that SingleBenchmarkCanBeExecutedForMultipleRuntimes takes more than 30 minutes.

**Comment by Ky7m** [comment]
@adamsitnik, I am done with changes and disabled xunit feature to avoid parallelization of test runs.

**Comment by Ky7m** [comment]
@adamsitnik let me know if I can try to fix CustomCoreClrToolchainTests.

**Comment by Ky7m** [comment]
Closes #875.

**Comment by AndreyAkinshin** [comment]
@Ky7m @adamsitnik can we merge this PR? Or do we have any additional issues that should be resolved first?

**Comment by adamsitnik** [comment]
@Ky7m sorry for the delay!

**Comment by AndreyAkinshin** [comment]
@Ky7m I don't see any builds here: https://dev.azure.com/dotnet/BenchmarkDotNet/_build
Should we do any additional steps to make it work?

**Comment by Ky7m** [comment]
@AndreyAkinshin it was configured, need to check if triggers work. Run manually https://dev.azure.com/dotnet/BenchmarkDotNet/_build

**Pull request merged**